datafusion/optimizer/src/analyzer/count_wildcard_rule.rs

```diff
@@ -97,19 +98,47 @@ fn analyze_internal(plan: LogicalPlan) -> Result<Transformed<LogicalPlan>> {
                 fetch,
             })))
         }
-        LogicalPlan::Projection(projection) => {
-            let projection_expr = projection
-                .expr
-                .iter()
-                .map(|expr| expr.clone().rewrite(&mut rewriter))
-                .collect::<Result<Vec<_>>>()?;
+        LogicalPlan::Projection(projection)
+            if matches!(
+                projection.input.as_ref(),
+                LogicalPlan::Aggregate(..) | LogicalPlan::Window(..)
+            ) =>
+        {
+            let (new_exprs, new_fields): (Vec<_>, Vec<_>) =
+                izip!(projection.expr.iter(), projection.schema.fields().iter())
+                    .map(|(expr, field)| {
+                        let new_field = match expr {
+                            Expr::Alias(_, _) => field.clone(),
+                            _ => {
+                                let mut name = field.field().name().clone();
+                                if name.contains(COUNT_STAR) {
+                                    name = name.replace(
+                                        COUNT_STAR,
+                                        count(lit(COUNT_STAR_EXPANSION))
+                                            .to_string()
+                                            .as_str(),
+                                    );
+                                }
+                                DFField::new(
+                                    field.qualifier().cloned(),
+                                    &name,
+                                    field.data_type().clone(),
+                                    field.is_nullable(),
+                                )
+                            }
+                        };
```

Comment:
It isn't a good method to resolve this issue.
IMO, it looks like a `hack` patch to fix issue.

Reply:
Yes, this piece of code is indeed difficult to understand and seems strange. However, I haven't found a good solution for it yet.

thanks for your help.

Reply:
i did some change.hope you have time for review.
thanks again.